### PR TITLE
#153356848 Add notes controller

### PR DESCRIPTION
**Pull request by Mnickii**
#### What does this PR do?

Add notes controller to power the note API routes.

#### Description of Task to be completed?

- Add notes API routes
- Include notes on GET incidents
- Add a notes controller with all the route methods

#### Where should the reviewer start?
..controllers/notes.js
#### How should this be manually tested?
```npm run start:dev ```, then use the notes API routes on postman
#### Any background context you want to provide?
Notes belong to Incidents
#### What are the relevant pivotal tracker stories?
#153356848

### CODE REVIEW

File: server/routes/index.js [hunk 1]
```diff
@@ -33,4 +34,12 @@ app.get('/api/levels', levelsService.list);
 app.get('/api/levels/:id', levelsService.findById);
 app.put('/api/levels/:id', levelsService.update);
 
+// notes endpoints
+app.post('/api/incidents/:id', notesService.create);
```
Comment (Mercy-Muchai) [inline server/routes/index.js]: Could we have this endpoint specifying on the route that its POSTing a note?`/api/incidents/:id/notes`

File: server/controllers/incidents.js [hunk 2]
```diff
@@ -28,7 +29,12 @@ module.exports = {
   // get all incidents
   list(req, res) {
     return Incident
-      .findAll()
+      .findAll({
```
Comment (Mercy-Muchai) [inline server/controllers/incidents.js]: 💯

File: server/controllers/incidents.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
 const Incident = require("../models").Incidents;
 const User = require("../models").Users;
+const Note = require("../models").Notes;
```
Comment (Mbarak-Mbigo) [inline server/controllers/incidents.js]: Why are we using the singular form instead of the plural form? `Notes`

Comment (Mnickii) [reply server/controllers/incidents.js]: The methods in each controller define what services a model uses. This is why we define the constant in singular form to reference the model. For example; the method `create` is meant to serve as a POST for a Note vs Notes. This is the convention we are using.

Comment (Mbarak-Mbigo) [reply server/controllers/incidents.js]: 👍

**Pull request merged**